### PR TITLE
feat: Fixes resolving of paths and bugs related to finding testdata

### PR DESCRIPTION
**Pull request by lfvbarcus**
This MR corrects the logic with prepending paths if the user has specified them in the reqstool_config. It also adresses an spelling error and the logic in the handle_custom() function that rewrote the default paths of maven and python projects to an empty list.

### CODE REVIEW

File: src/reqstool/requirements_indata/requirements_indata_paths.py [hunk 1]
```diff
@@ -40,12 +41,23 @@ def prepend_paths(self, prepend_dir):
...
-                and isinstance(getattr(self, prop), RequirementsIndataPathItem)
+                and self.attribute_is_instance_or_list_of_class(prop_name=prop, the_class=RequirementsIndataPathItem)
                 and prop not in ["requirements_yml", "svcs_yml", "mvrs_yml"]
             ):
                 path_item = getattr(self, prop)
-                if path_item.path is not None:
-                    path_item.path = str(PurePath(prepend_dir, path_item.path))
+                self.check_type_and_prepend_path(path_item=path_item, prepend_dir=prepend_dir)
+
+    # prepend path for each entry in a list of RequirementsIndataPathItem(s) or just on a single entity of the class
```
Comment (lfvjimisola) [inline src/reqstool/requirements_indata/requirements_indata_paths.py:50]: if there is a need to comment the code then document according to our internal development manual

File: src/reqstool/requirements_indata/requirements_indata_paths.py [hunk 2]
```diff
@@ -40,12 +41,23 @@ def prepend_paths(self, prepend_dir):
...
+    def check_type_and_prepend_path(self, path_item, prepend_dir: str):
+        if type(path_item) is list:
+            for entry in path_item:
+                if type(entry) is RequirementsIndataPathItem and entry.path is not None:
```
Comment (lfvjimisola) [inline src/reqstool/requirements_indata/requirements_indata_paths.py:54]: Can this happen? We are parsing into RequirementsIndataPathItem right? I don't think we shall just log and continue. There is a serious issue with the parsing if this fails.

Comment (lfvjimisola) [reply src/reqstool/requirements_indata/requirements_indata_paths.py:54]:
```python
def check_type_and_prepend_path(self, path_item, prepend_dir: str):
    if isinstance(path_item, list):
        if all(isinstance(entry, RequirementsIndataPathItem) and entry.path is not None for entry in path_item):
            path_item = [entry.path = str(PurePath(prepend_dir, entry.path)) for entry in path_item]
    else:
        if path_item.path is not None:
            path_item.path = str(PurePath(prepend_dir, path_item.path))

```

Comment (lfvbarcus) [reply src/reqstool/requirements_indata/requirements_indata_paths.py:54]: This is really just a case for the test_result_dirs as they are expressed as a list of RequirementIndataPathItems

For example, the MavenDefault Paths

image


without this logic, the paths for those objects are not handled correctly.

Comment (lfvbarcus) [reply src/reqstool/requirements_indata/requirements_indata_paths.py:54]: However, I agree that just a log in not robust enough to handle this. I could throw an exception as well.

File: src/reqstool/requirements_indata/requirements_indata_paths.py (hunk 1 above)
Comment (lfvbarcus) [reply src/reqstool/requirements_indata/requirements_indata_paths.py]: fixed!

**Review by lfvjimisola — APPROVED** [review]

**Pull request merged**